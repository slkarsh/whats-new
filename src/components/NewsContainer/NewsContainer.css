main {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    margin-top: 80px;
    width: 100%;
    height: 100vh;
    overflow: scroll;
}