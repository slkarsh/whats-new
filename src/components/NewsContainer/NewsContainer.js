import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle'

// NEWSCONTAINER COMPONENT CODE GOES HERE

const NewsContainer = props => {
    const localNews = props.local.map(localNews => {
        return (
            <NewsArticle
                key={localNews.id}
                headline={localNews.headline}
                description={localNews.description}
                imgSrc={localNews.img}
                link={localNews.url}
             />
        )
    })
    return (
        <main className='NewsContainer'>
            {localNews}
        </main>
    )
}

export default NewsContainer;