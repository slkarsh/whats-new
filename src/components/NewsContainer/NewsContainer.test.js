import React from 'react'
import { shallow } from 'enzyme'
import NewsContainer from './NewsContainer'

describe('NewsContainer', () => {
    it('should match the snapshot when rendered correctly', () => {
        let mockCurrent = [
        {id: 1, headline:'headline 1', description: 'description 1', img: 'woof', url: 'woof dot com'},
        {id: 2, headline: 'headline 2', description: 'description 2', img: 'meow', url: 'meow dot com'} 
        ]
        const wrapper = shallow(<NewsContainer
            current = {mockCurrent}
             />)
        expect(wrapper).toMatchSnapshot()
    })
})