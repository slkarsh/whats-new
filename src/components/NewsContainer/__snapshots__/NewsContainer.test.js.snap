// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the snapshot when rendered correctly 1`] = `
<main
  className="NewsContainer"
>
  <NewsArticle
    description="description 1"
    headline="headline 1"
    imgSrc="woof"
    key="1"
    link="woof dot com"
  />
  <NewsArticle
    description="description 2"
    headline="headline 2"
    imgSrc="meow"
    key="2"
    link="meow dot com"
  />
</main>
`;
